projects/dev/src/app/group/group-manage/pending-request/pending-request.component.ts

```diff
@@ -1,108 +1,190 @@
+    const succ = _.countBy(result.data, (status: string) => {
+      return ["success", "unchanged"].includes(status);
+    });
 
-  _setRequestData(reqs: PendingRequest[]) {
-    this.requests = [];
-
-    for (const req of reqs) {
-      this.requests.push({
-        member_id: req.member_id,
-        "joining_user.login": `${req.joining_user.first_name || ""} ${
-          req.joining_user.last_name || ""
-        } (${req.joining_user.login || ""})`,
-        "grade": req.joining_user && req.joining_user.grade ? req.joining_user.grade : null,
-        at: req.at,
+    if (succ.false === undefined || succ.false === 0) {
+      this.messageService.add({
+        severity: "success",
+        summary: "Success",
+        detail: `${succ.true} request(s) have been ${msg}`,
+        life: TOAST_LENGTH,
+      });
+    } else if (succ.true === undefined || succ.true === 0) {
+      this.messageService.add({
+        severity: "error",
+        summary: "Error",
+        detail: `Unable to ${verb} the selected request(s).`,
+        life: TOAST_LENGTH,
+      });
+    } else {
+      this.messageService.add({
+        severity: "warn",
+        summary: "Partial success",
+        detail: `${succ.true} request(s) have been ${msg}, ${succ.false} could not be executed`,
+        life: TOAST_LENGTH,
       });
     }
   }
 
-  constructor(private groupService: GroupService) {}
+  _processRequestError(_err) {
+    this.messageService.add({
+      severity: "error",
+      summary: "Error",
+      detail: ERROR_MESSAGE.fail,
+      life: TOAST_LENGTH,
+    });
+  }
+
+  isAccepting() {
+    return this.onGoingActivity === Activity.Accepting;
+  }
+
+  isRejecting() {
+    return this.onGoingActivity === Activity.Rejecting;
+  }
+
+  isIdle() {
+    return this.onGoingActivity === Activity.None;
+  }
+
+  constructor(
+    private groupService: GroupService,
+    private messageService: MessageService
+  ) {}
 
   ngOnInit() {
     this.panel.push({
       name: "Pending Requests",
       columns: this.columns,
     });
-    this.groupService
-      .getManagedRequests(this.id)
-      .subscribe((reqs: PendingRequest[]) => {
-        this._setRequestData(reqs);
-      });
   }
 
-  ngOnChanges(changes: SimpleChanges) {
-    this.groupService
-      .getManagedRequests(this.id)
-      .subscribe((reqs: PendingRequest[]) => {
-        this._setRequestData(reqs);
-      });
+  ngOnChanges(_changes: SimpleChanges) {
+    this.selection = [];
+    this._reloadData();
   }
 
-  onExpandWidth(e) {}
+  onAcceptOrReject(action: Action) {
+    if (this.selection.length === 0 || this.onGoingActivity !== Activity.None) {
+      return;
+    }
+
+    let resultObserver: Observable<RequestActionResponse>;
+    this.onGoingActivity = (action === Action.Accept) ? Activity.Accepting : Activity.Rejecting;
+    
+    const group_ids = this.selection.map((req: PendingRequest) => req.joining_user.group_id);
```

Comment:
Just note (no need to change) that when I insist on everything to be typed, it is for compiler/transpiler/linter to be able to check it. Here, as `this.selection` is already typed, the map does not need to have its arg typed as TS knows implicitely (and your IDE can tell you) that it is a PendingRequest... it would allow to shorter/more readable code. In the same way, we could write `....map<string>(...)` but it would be redundant.. 
Usually, it is considered that derived types do not have to be written (in some case, it can clarify code though).

Reply:
fixed.